src/DataTransferObject.php

```diff
@@ -24,12 +24,12 @@ abstract class DataTransferObject implements DtoContract
     /** @var bool */
     protected $immutable;
 
-    public static function mutable(array $parameters): DtoContract
+    public static function mutable(array $parameters): self
```

Comment:
I'm unsure whether the `mutable` static constructor should be added. I find this confiusing:

```php
$dto = MyDTO::mutable(/* … */);

// Won't work, even though we made a mutable object
$dto->foo = 'bar'
```

_If_ we'd keep this construct, I'd expect it to make the whole object mutable. But I don't think we want to allow that.

Reply:
removed mutable method